saleor/data_feeds/google_merchant.py

```diff
@@ -0,0 +1,219 @@
+    else:
+        return None
+
+
+def item_availability(item):
+    if item.get_stock_quantity():
+        return 'in stock'
+    else:
+        return 'out of stock'
+
+
+def item_shipping(item):
+    """
+    You can set one shipping cost in feed settings or precise shipping cost
+    for every item.
+    Read more:
+    https://support.google.com/merchants/answer/7050921
+    """
+    return ''
+
+
+def item_google_product_category(item, category_paths):
+    """
+    To have your categories accepted, please use names accepted by Google or
+    write custom function which maps your category names into to Google codes.
+    Read more:
+    https://support.google.com/merchants/answer/6324436
+    """
+    category = item.product.get_first_category()
+    if category:
+        if category.pk in category_paths:
+            return category_paths[category.pk]
+        ancestors = [ancestor.name for ancestor
+                     in list(category.get_ancestors())]
+        category_path = CATEGORY_SEPARATOR.join(ancestors + [category.name])
+        category_paths[category.pk] = category_path
+        return category_path
+    else:
+        return ''
+
+
+def item_price(item):
+    price = item.get_price_per_item(discounts=None)
+    return '%s %s' % (price.gross, price.currency)
+
+
+def item_sale_price(item, discounts):
+    sale_price = item.get_price_per_item(discounts=discounts)
+    return '%s %s' % (sale_price.gross, sale_price.currency)
+
+
+def item_attributes(item, categories, category_paths, current_site,
+                    discounts):
+    product_data = {
+        'id': item_id(item),
+        'title': item_title(item),
+        'description': item_description(item),
+        'condition': item_condition(item),
+        'mpn': item_mpn(item),
+        'item_group_id': item_group_id(item),
+        'availability': item_availability(item),
+        'google_product_category': item_google_product_category(
+            item, category_paths),
+        'link': item_link(item, current_site),
+        'shipping': item_shipping(item),
+        'brand': item_brand(item)
+    }
+
+    image_link = item_image_link(item, current_site)
+    if image_link:
+        product_data['image_link'] = image_link
+
+    price = item_price(item)
+    product_data['price'] = price
+    sale_price = item_sale_price(item, discounts)
+    if sale_price != price:
+        product_data['sale_price'] = sale_price
+
+    tax = item_tax(item, discounts)
+    if tax:
+        product_data['tax'] = tax
+
+    shipping = item_shipping(item)
+    if shipping:
+        product_data['shipping'] = shipping
+
+    brand = item_brand(item)
+    if brand:
+        product_data['brand'] = brand
+
+    return product_data
+
+
+def update_feed():
+    with default_storage.open(FILE_PATH, 'wb') as output_file:
+        if COMPRESSION:
+            try:
+                output = gzip.open(output_file, 'wt')
+            except TypeError:
```

Comment:
I feel like this can be determined at import time as it's unlikely to change in runtime.